app/lib/LevelLoader.coffee

```diff
@@ -221,36 +220,38 @@ module.exports = class LevelLoader extends CocoClass
         @worldNecessities.push heroResource
       @sessionDependenciesRegistered[session.id] = true
     unless @level.isType('hero', 'hero-ladder', 'hero-coop')
+      Backbone.Mediator.publish 'level:session-loaded', level: @level, session: @session
       # Return before loading heroConfig ThangTypes. Finish if all world necessities were completed by the time the session loaded.
       if @checkAllWorldNecessitiesRegisteredAndLoaded()
         @onWorldNecessitiesLoaded()
-      return
-    # Load the ThangTypes needed for the session's heroConfig for these types of levels
-    heroConfig = session.get('heroConfig')
-    heroConfig ?= me.get('heroConfig') if session is @session and not @headless
-    heroConfig ?= {}
-    heroConfig.inventory ?= feet: '53e237bf53457600003e3f05'  # If all else fails, assign simple boots.
-    heroConfig.thangType ?= '529ffbf1cf1818f2be000001'  # If all else fails, assign Tharin as the hero.
-    session.set 'heroConfig', heroConfig unless _.isEqual heroConfig, session.get('heroConfig')
-    url = "/db/thang.type/#{heroConfig.thangType}/version"
-    if heroResource = @maybeLoadURL(url, ThangType, 'thang')
-      @worldNecessities.push heroResource
     else
-      heroThangType = @supermodel.getModel url
-      @loadDefaultComponentsForThangType heroThangType
-      @loadThangsRequiredByThangType heroThangType
-
-    for itemThangType in _.values(heroConfig.inventory)
-      url = "/db/thang.type/#{itemThangType}/version?project=name,components,original,rasterIcon,kind"
-      if itemResource = @maybeLoadURL(url, ThangType, 'thang')
-        @worldNecessities.push itemResource
+      # Load the ThangTypes needed for the session's heroConfig for these types of levels
+      heroConfig = session.get('heroConfig')
+      heroConfig ?= me.get('heroConfig') if session is @session and not @headless
+      heroConfig ?= {}
+      heroConfig.inventory ?= feet: '53e237bf53457600003e3f05'  # If all else fails, assign simple boots.
+      heroConfig.thangType ?= '529ffbf1cf1818f2be000001'  # If all else fails, assign Tharin as the hero.
+      session.set 'heroConfig', heroConfig unless _.isEqual heroConfig, session.get('heroConfig')
+      Backbone.Mediator.publish 'level:session-loaded', level: @level, session: @session
```

Comment:
Should this also only happen if `@session is session`? Did you test when playing ladder levels?

Reply:
Yikes, good catch.  No, I didn't test ladder levels.  Going to update this logic to be parity with existing behavior, except delay session-loaded if we're going to set heroConfig too.